### PR TITLE
refactor: use `Vector` in `Fixpoint/Ram`

### LINKED ISSUE #5484: Use `Vector` in Datalog engine
Each in their own PR:
- [ ] As a replacement for `List` in `Fixpoint/Ram`
- [ ] As the return type of `factsX` in `Solver.flix`.
- [ ] As a replacement for `List` in `Fixpoint/Ast`

### PR DESCRIPTION
**Pull request by sockmaster27**
Replaces all instances of `List` with `Vector` in `Fixpoint/Ram`, and edits places where these types are used, avoiding conversions wherever possible. 

Fixes #5484

### CODE REVIEW

File: main/src/library/Fixpoint/Interpreter.flix [hunk 1]
```diff
@@ -92,15 +92,15 @@ mod Fixpoint {
...
+                        let tuple = Vector.map(evalTerm(env), terms);
                         MutMap.put!(tuple, Reflect.default(), rel)
                     case Denotation.Latticenal(bot, leq, lub, _) =>
                         // assume that length(terms) > 0
-                        let len = List.length(terms);
-                        let keyList = terms |> List.map(evalTerm(env));
-                        let (relKeys, latValList) = List.splitAt(len-1, keyList);
-                        let key = List.toVector(relKeys);
-                        let latVal = match List.head(latValList) {
+                        let len = Vector.length(terms);
```
Comment (magnus-madsen) [inline main/src/library/Fixpoint/Interpreter.flix:99]: Is this because Vector lacks a `splitAt`?

Comment (sockmaster27) [reply main/src/library/Fixpoint/Interpreter.flix:99]: Yes

Comment (sockmaster27) [reply main/src/library/Fixpoint/Interpreter.flix:99]: Sure!

Comment (magnus-madsen) [reply main/src/library/Fixpoint/Interpreter.flix:99]: Ok, let us do that and then rebase.

**Comment by magnus-madsen** [comment]
Conflicted.

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**